src/controllers/auth.controller.js

```diff
@@ -5,28 +5,112 @@ import { hashPassword } from '../utils/crypto'
 import db from '../models/index'
 import passport from 'passport'
 import jwt from 'jsonwebtoken'
-
+import {
+  validateEmail,
+  sendEmail,
+  checkPassword,
+} from '../services/authService'
+import { google } from 'googleapis'
+require('dotenv').config()
+const { OAuth2 } = google.auth
+const client = new OAuth2(process.env.LOGIN_GOOGLE_CLIENT_ID)
 @controller('/api/auth')
 class AuthCtrl extends BaseCtrl {
   @post('/register')
   async register(req, res) {
     const { username, email, password, firstName, lastName } = req.body
     const hash = await hashPassword(password)
+    //Validate and check acc
+    if (!validateEmail(email)) {
+      return res.status(400).json({ success: false, message: 'Invalid email' })
+    }
+    const checkUsername = await db.User.findOne({
+      where: { username: username },
+    })
+    const checkEmailActive = await db.User.findOne({
+      where: { email: String(email).toLowerCase(), status: 'active' },
+    })
+    const checkEmailPending = await db.User.findOne({
+      where: { email: String(email).toLowerCase(), status: 'pending' },
+    })
+    if (checkUsername) {
+      return res
+        .status(400)
+        .json({ success: false, message: 'Username already exists. ' })
+    }
+    if (checkEmailActive) {
+      return res
+        .status(400)
+        .json({ success: false, message: 'Email already exists. ' })
+    }
+    if (checkEmailPending) {
+      return res.status(400).json({
+        success: false,
+        message: 'Email was not active, please check your email to activate. ',
+      })
+    }
+    if (checkPassword(password)) {
+      return res.status(400).json({
+        success: false,
+        message: 'Password must at least 6 characters.',
+      })
+    }
     try {
       const newUser = await db.User.create({
         username,
         email,
         password: hash,
         firstName,
         lastName,
+        status: 'pending',
```

Comment:
Can we create a constant.js file and export theses string from it
I think it is better to turn to upper case string for constant pending -> PENDING

Reply:
Yeah, I was created a constant.js file in folder utils